Bottom: 46daf55eeac1f207c16db23be33bcc434cfd58e6
Top:    4140bdfdea0385175479fa47feb983023ef59155
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-27 17:59:53 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6264a6f..eacb418 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1123,7 +1123,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
